src/JsonSchema/PatternKeyword.cs

```diff
@@ -23,39 +24,44 @@ namespace Json.Schema;
 [JsonConverter(typeof(PatternKeywordJsonConverter))]
 public class PatternKeyword : IJsonSchemaKeyword
 {
+
+
+	private readonly RegexOrPattern _regexOrPattern;
+
 	/// <summary>
 	/// The JSON name of the keyword.
 	/// </summary>
 	public const string Name = "pattern";
 
+	public string Pattern => _regexOrPattern;
+
 	/// <summary>
-	/// The regular expression.
-	/// </summary>
-	public Regex Value { get; }
-	/// <summary>
-	/// If the pattern is invalid or unsupported by <see cref="Regex"/>, it will appear here.
+	/// Returns the Regex Value of the keyword
 	/// </summary>
-	/// <remarks>
-	/// All validations will fail if this is populated.
-	/// </remarks>
-	public string? InvalidPattern { get; }
```

Comment:
Same here.

Reply:
I've added a couple tests to `master` and rebased to show this.

Reply:
If `RegexOrPattern` were to check the pattern in the constructor, would that insert the pattern into the Regex cache (what this whole change was trying to do anyway)?

> I would say that it's an illegal json schema when the regex pattern is invalid, so an exception when deserializing json to a JsonSchema sounds valid.

I want to say there were some optional tests that had valid regexes that .Net just didn't support.  It seems the only one that fits that now are the UTF-16 tests, but even those just don't evaluate correctly.

Moreover, the `.Pattern()` builder extension _does_ throw an exception on a bad pattern as well.

I think I want to change the behavior long term, but I'd still like to leave the current support for now.  I'll work something out.

Reply:
I suppose I can include this change and say in the release notes that it's fixing a bug.  Doesn't really explain why the `InvalidPattern` property was there in the first place, though.

Reply:
Yeah, performing an `IsMatch` in the constructor works as well, and is actually closer to the original behavior.